src/utils.ts

```diff
@@ -186,6 +187,9 @@ export const isPromiseLike = <T>(
 
 type AsyncFunctionType<A extends unknown[], R> = (...args: A) => PromiseLike<R>
 
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const AsyncFunctionConstructor = (async () => { }).constructor as FunctionConstructor
+
```

Comment:
this is better, as we basically cache the constructor avoiding instantiating empty async function for every async Function check